Consider some form of static analysis in CI

Following #59 we ought to extend our integration checks to cover oversights:

```console
…/mkdocs-drawio-exporter on  HEAD (157dd90) is 📦 v0.9.0 via 🐍 v3.12.2
at 22:53:02 ✖ 1 🐠 ❯ asdf shell python 3.8.19

…/mkdocs-drawio-exporter on  main [⇣1] is 📦 v0.9.0 via 🐍 v3.8.19 took 3s 
at 22:53:12 🐠 ❯ g sw --detach v0.9.0
HEAD is now at 157dd90 Bump version to v0.9.0

…/mkdocs-drawio-exporter on  HEAD (157dd90) is 📦 v0.9.0 via 🐍 v3.8.19 
at 22:53:17 🐠 ❯ python -m flake8 mkdocs_drawio_exporter/
mkdocs_drawio_exporter/__init__.py:1:1: F401 '.exporter.DrawIoExporter' imported but unused
mkdocs_drawio_exporter/__init__.py:1:1: F401 '.exporter.Source' imported but unused
mkdocs_drawio_exporter/__init__.py:2:1: F401 '.plugin.DrawIoExporterPlugin' imported but unused
mkdocs_drawio_exporter/exporter.py:271:62: E999 SyntaxError: invalid syntax
mkdocs_drawio_exporter/plugin.py:52:61: E999 SyntaxError: invalid syntax
mkdocs_drawio_exporter/tests/__init__.py:1:1: F401 '.exporter.ExporterTests' imported but unused
mkdocs_drawio_exporter/tests/exporter.py:82:80: E501 line too long (95 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:108:80: E501 line too long (87 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:129:80: E501 line too long (109 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:134:80: E501 line too long (96 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:141:80: E501 line too long (82 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:144:80: E501 line too long (90 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:161:80: E501 line too long (113 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:191:28: E711 comparison to None should be 'if cond is None:'
mkdocs_drawio_exporter/tests/exporter.py:207:80: E501 line too long (113 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:219:28: E711 comparison to None should be 'if cond is None:'
mkdocs_drawio_exporter/tests/exporter.py:236:80: E501 line too long (113 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:269:80: E501 line too long (108 > 79 characters)
mkdocs_drawio_exporter/tests/exporter.py:286:23: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
mkdocs_drawio_exporter/tests/exporter.py:325:23: E711 comparison to None should be 'if cond is None:
```

--- a/.envrc
+++ b/.envrc
@@ -1,1 +1,0 @@
-use flake

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -81,7 +81,11 @@ jobs:
           cache: poetry
           python-version: "${{ matrix.python }}"
 
+      - name: Install dependencies
+        run: poetry install --with dev
+
       - name: Run tests
-        run: |
-          poetry install
-          poetry run test
+        run: poetry run test
+
+      - name: Lint
+        run: poetry run ruff check

--- a/README.md
+++ b/README.md
@@ -121,7 +121,7 @@ To get completion working in your editor, set up a virtual environment in the ro
 ```shell
 pip3 install --user --upgrade wheel
 pipx install twine
-poetry install
+poetry install --with dev
 ```
 
 To install the plugin onto a local MkDocs site in editable form:

--- a/README.md
+++ b/README.md
@@ -116,11 +116,21 @@ plugins:
 
 ## Hacking
 
+Optionally, use [Nix](https://nix.dev/) for a development shell with all the necessary dependencies:
+
+```shell
+nix develop
+```
+
+We use [Poetry](https://python-poetry.org/) for dependency management:
+
+```shell
+poetry install
+```
+
 To get completion working in your editor, set up a virtual environment in the root of this repository and install MkDocs:
 
 ```shell
-pip3 install --user --upgrade wheel
-pipx install twine
 poetry install --with dev
 ```
 

--- a/flake.nix
+++ b/flake.nix
@@ -7,18 +7,22 @@
   };
 
   outputs = { flake-utils, nixpkgs, ... }:
-  flake-utils.lib.eachDefaultSystem (system: {
+  flake-utils.lib.eachDefaultSystem (system: rec {
     devShells.default =
       let
         pkgs = import nixpkgs { inherit system; };
-      in
-        pkgs.mkShell {
-          packages = with pkgs; [
-            drawio
-            poetry
-            python312
-            python312Packages.python-lsp-server
+        fhs = pkgs.buildFHSUserEnv {
+          name = "fhs-shell";
+          targetPkgs = pkgs: [
+            pkgs.drawio
+            pkgs.nil
+            pkgs.poetry
+            pkgs.python312
+            pkgs.python312Packages.python-lsp-server
           ];
+          runScript = "bash";
         };
+      in
+        fhs.env;
   });
 }

--- a/flake.nix
+++ b/flake.nix
@@ -13,7 +13,12 @@
         pkgs = import nixpkgs { inherit system; };
       in
         pkgs.mkShell {
-          packages = with pkgs; [ python313 python312Packages.python-lsp-server ];
+          packages = with pkgs; [
+            drawio
+            poetry
+            python312
+            python312Packages.python-lsp-server
+          ];
         };
   });
 }

--- a/mkdocs_drawio_exporter/__init__.py
+++ b/mkdocs_drawio_exporter/__init__.py
@@ -1,2 +1,4 @@
+__all__ = ['DrawIoExporter', 'DrawIoExporterPlugin', 'Source']
+
 from .exporter import DrawIoExporter, Source
 from .plugin import DrawIoExporterPlugin

--- a/mkdocs_drawio_exporter/exporter.py
+++ b/mkdocs_drawio_exporter/exporter.py
@@ -383,5 +383,5 @@ class DrawIoExporter:
         try:
             self.log.debug(f'Using export command {cmd}')
             return subprocess.call(cmd)
-        except:
+        except (OSError, subprocess.CalledProcessError):
             self.log.exception('Subprocess raised exception')

--- a/mkdocs_drawio_exporter/plugin.py
+++ b/mkdocs_drawio_exporter/plugin.py
@@ -7,7 +7,7 @@ import mkdocs.plugins
 from mkdocs.structure.files import Files
 from mkdocs.utils import copy_file
 
-from .exporter import ConfigurationError, DrawIoExporter, Configuration
+from .exporter import ConfigurationError, DrawIoExporter
 
 
 log = mkdocs.plugins.log.getChild('drawio-exporter')

--- a/mkdocs_drawio_exporter/tests/__init__.py
+++ b/mkdocs_drawio_exporter/tests/__init__.py
@@ -1,1 +1,3 @@
+__all__ = ['ExporterTests']
+
 from .exporter import ExporterTests

--- a/mkdocs_drawio_exporter/tests/exporter.py
+++ b/mkdocs_drawio_exporter/tests/exporter.py
@@ -188,7 +188,7 @@ class ExporterTests(unittest.TestCase):
         _, exit_status = exporter.ensure_file_cached(
                 source, source_rel, 0, config)
 
-        assert exit_status == None
+        assert exit_status is None
         self.log.warning.assert_called_once()
 
     def test_ensure_file_cached_skips_export_if_cache_fresh(self):
@@ -216,7 +216,7 @@ class ExporterTests(unittest.TestCase):
                 source, source_rel, 0, config)
 
         assert cache_filename == exporter.make_cache_filename.return_value
-        assert exit_status == None
+        assert exit_status is None
         exporter.use_cached_file.assert_called_once()
         assert not exporter.export_file.called
 
@@ -283,7 +283,7 @@ class ExporterTests(unittest.TestCase):
         getmtime_mock.side_effect = getmtime_return_value
 
         result = exporter.use_cached_file(source, cache_filename)
-        assert result == False
+        assert not result
 
     @patch('subprocess.call')
     def test_export_file(self, call_mock):
@@ -322,7 +322,7 @@ class ExporterTests(unittest.TestCase):
         result = exporter.export_file(
                 source, 0, dest, config)
 
-        assert result == None
+        assert result is None
         self.log.exception.assert_called_once()
         call_mock.assert_called_once()
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -44,6 +44,12 @@ pyyaml = ">=6.0.1"
 six = ">=1.16.0"
 tornado = ">=6.3.3"
 
+[tool.poetry.group.dev]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+ruff = ">=0.7.2"
+
 [tool.poetry.urls]
 homepage = "https://github.com/LukeCarrier/mkdocs-drawio-exporter"
 repository = "https://github.com/LukeCarrier/mkdocs-drawio-exporter"

